lib/sdf-server/src/server/service/component/update_property_editor_value.rs

```diff
@@ -78,3 +78,26 @@ pub async fn update_property_editor_value(
     }
     Ok(response.body(axum::body::Empty::new())?)
 }
+
+async fn update_property_editor_value_inner(
+    ctx: &DalContext,
+    attribute_value_id: AttributeValueId,
+    maybe_value: Option<serde_json::Value>,
+    is_for_secret: bool,
+) -> ComponentResult<()> {
+    // If this is for a secret and we have a provided value, then we need to use the attach method.
+    // Otherwise, we can proceed with a regular attribute value update. We can do this because if
+    // there's no value provided, then "detaching" the secret works the same way as "unsetting"
+    // an attribute value with an empty value.
+    if let Some(value) = maybe_value.as_ref() {
+        if is_for_secret {
+            let secret_id: SecretId = serde_json::from_value(value.to_owned())?;
+            Secret::attach_for_attribute_value(ctx, attribute_value_id, Some(secret_id)).await?;
+            return Ok(());
+        }
+    }
+
+    // Regardless of whether this is for a secret, unsetting works the same way.
+    AttributeValue::update(ctx, attribute_value_id, maybe_value).await?;
+    Ok(())
+}
```

Comment:
We will also want to align this with how we handle "unsetting" in the attach method.